### PR TITLE
Docs/datasources usman

### PR DESCRIPTION
**Pull request by usmangt**
This PR fixes the following 2 pages:

For Link 1: https://grafana.com/docs/grafana/latest/administration/data-source-management/

It will fix the labels/tags and only keep the content that is exclusive to Grafana Cloud/Enterprise. 

Also fixes some minor tweaks e.g. spell check, paragraphing, bullets etc.

For Link 2: https://grafana.com/docs/grafana/latest/datasources/

It adds the section from Link 1 which is NOT exclusive to Grafana Cloud/Enterprise. That means this link will have all the general information that is available in all Grafana flavors (OSS. Cloud and Enterprise)

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

Signed-off-by: Syed Usman Ahmad <usman.ahmad@grafana.com>

### CODE REVIEW

**Review by chri2547 — COMMENTED** [review]
This looks ok to me. Would like @lwandz13  to also review. And @jdbaldry , the doc validator doesn't like `add-a-data-source` in the aliases. Not sure how to resolve that one. Thanks, @usmangt

File: docs/sources/administration/data-source-management/_index.md
Comment (jdbaldry) [inline docs/sources/administration/data-source-management/_index.md:32]:
```suggestion
- The `query` permission allows users to query the data source.
- The `edit` permission allows users to query the data source, edit the data source’s configuration and delete the data source.
- The `admin` permission allows users to query and edit the data source, change permissions on the data source and enable or disable query caching for the data source.
```

File: docs/sources/datasources/_index.md
Comment (jdbaldry) [inline docs/sources/datasources/_index.md:117]: This link will need to be updated into a `docs/reference` style link like the others in the page because the linked "Administration" content doesn't exist in Grafana Cloud documentation.

For an example, refer to https://github.com/grafana/grafana/pull/83712/files#diff-7b86f2d305e1ba40bef37131931d4213ee72232f66190c6a31cc1de8b4d913ffR129-R130.

File: docs/sources/datasources/_index.md
Comment (jdbaldry) [inline docs/sources/datasources/_index.md:120]: This link will also need to be updated into a `docs/reference` style link like the others in the page because the linked "Administration" content doesn't exist in Grafana Cloud documentation.

For an example, refer to https://github.com/grafana/grafana/pull/83712/files#diff-7b86f2d305e1ba40bef37131931d4213ee72232f66190c6a31cc1de8b4d913ffR129-R130.

File: docs/sources/datasources/_index.md
Comment (jdbaldry) [inline docs/sources/datasources/_index.md:126]: I think the link text should not have quotes and we should avoid first person generally unless we've previously explicitly referred to our organization.

```suggestion
To build your own data source plugin, refer to the [Build a data source plugin](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and [Plugin tools](/developers/plugin-tools).
```

**Review by jdbaldry — COMMENTED** [review]
Yeah, this page previously had a heading "Add a data source" that was linked to by other pages. We should update those links to go somewhere more appropriate. Give me a ping if you need help finding alternatives.

**Comment by jdbaldry** [comment]
> Would like @lwandz13 to also review

Larissa is no longer CODEOWNER for data sources so I cover it now.

File: docs/sources/datasources/_index.md
Comment (usmangt) [reply docs/sources/datasources/_index.md:120]: This now fixed, please check

**Review by usmangt — COMMENTED** [review]
fixed the link as suggested. please check

File: docs/sources/datasources/_index.md
Comment (jdbaldry) [reply docs/sources/datasources/_index.md:120]: The other is indeed fixed thanks for doing that :) Could you do the same with this one.

Comment (usmangt) [reply docs/sources/datasources/_index.md:120]: I am not sure which link you meant? is it about the Grafana Enterprise link?

Comment (jdbaldry) [reply docs/sources/datasources/_index.md:120]: Yeah that's the one. Sorry for the confusion, I copy-pasted my comment from the first instance and didn't update the link text.

Comment (usmangt) [reply docs/sources/datasources/_index.md]: This is now fixed !! :slightly_smiling_face:

**Review by lwandz13 — COMMENTED** [review]
this looks ok to me too...however Jack will need to do a final approval before this gets merged as I am no longer code owner for data sources.

**Review by jdbaldry — COMMENTED** [review]
Some of the files need formatting fixes: 
```
+ yarn run prettier:checkDocs
[warn] docs/sources/administration/data-source-management/_index.md
[warn] docs/sources/datasources/_index.md
[warn] Code style issues found in 2 files. Run Prettier to fix.
```

You can resolve it by following the instructions in https://github.com/grafana/technical-documentation/wiki/Software-workflows#lint-docs. If you have issues, I can run it over the files for you.

There are also two broken anchor links as a result of this change that need to be updated:

[doc-validator] reported by reviewdog 🐶
- Check failure on line 80 in docs/sources/getting-started/build-first-dashboard.md
  
  The anchor 'add-a-data-source' doesn't exist in the file 'docs/sources/administration/data-source-management/_index.md'.
Replace the anchor with one of the available anchors.

  Available anchors are: "data-source-management, data-source-permissions, assign-data-source-permissions-to-users-service-accounts-teams-or-roles, edit-data-source-permissions-for-users-service-accounts-teams-or-roles, remove-data-source-permissions-for-users-service-accounts-teams-or-roles, query-and-resource-caching, caching-benefits, data-sources-that-work-with-query-caching, enable-and-configure-query-caching, disable-query-caching, clear-cache, sending-a-request-without-cache"
- Check failure on line 41 in docs/sources/getting-started/get-started-grafana-influxdb.md

  The anchor 'add-a-data-source' doesn't exist in the file 'docs/sources/administration/data-source-management/_index.md'.
Replace the anchor with one of the available anchors.

  Available anchors are: "data-source-management, data-source-permissions, assign-data-source-permissions-to-users-service-accounts-teams-or-roles, edit-data-source-permissions-for-users-service-accounts-teams-or-roles, remove-data-source-permissions-for-users-service-accounts-teams-or-roles, query-and-resource-caching, caching-benefits, data-sources-that-work-with-query-caching, enable-and-configure-query-caching, disable-query-caching, clear-cache, sending-a-request-without-cache"

**Review by jdbaldry — APPROVED** [review]
LGTM!

**Pull request merged**
